### ISSUE TITLE
Error executing 'postInstallation': EACCES: permission denied, mkdir '/bitnami/wordpress/wp-content'

### ISSUE THREAD

**Issue by Neikius** [bitnami/bitnami-docker-wordpress#250]
**Description**
Trying to run wordpress and mounting the volume as host directory.
Tried this on 2 different systems (ubuntu 20.4 and tumbleweed). Pasting output from my workstation (TW).

**Steps to reproduce the issue:**
```
docker network create wordpress-network

docker volume create --name mariadb_data

docker run -d --name mariadb \
  --env ALLOW_EMPTY_PASSWORD=yes \
  --env MARIADB_USER=bn_wordpress \
  --env MARIADB_DATABASE=bitnami_wordpress \
  --network wordpress-network \
  --volume mariadb_data:/bitnami/mariadb \
  bitnami/mariadb:latest
  
docker run -d --name wordpress \
-p 8080:8080 -p 8443:8443 \
--env ALLOW_EMPTY_PASSWORD=yes \
--env WORDPRESS_DATABASE_USER=bn_wordpress \
--env WORDPRESS_DATABASE_NAME=bitnami_wordpress \
--network wordpress-network \
--volume /tmp/wp:/bitnami/wordpress \
bitnami/wordpress:latest
```

**Describe the results you received:**

```
nami    INFO  Initializing apache
nami    INFO  apache successfully initialized
nami    INFO  Initializing mysql-client
nami    INFO  mysql-client successfully initialized
nami    INFO  Initializing wordpress
wordpre INFO  ==> Preparing Varnish environment
wordpre INFO  ==> Preparing Apache environment
wordpre INFO  ==> Preparing PHP environment
mysql-c INFO  Trying to connect to MySQL server
mysql-c INFO  Found MySQL server listening at mariadb:3306
mysql-c INFO  MySQL server listening and working at mariadb:3306
wordpre INFO  Preparing WordPress environment
Error executing 'postInstallation': EACCES: permission denied, mkdir '/bitnami/wordpress/wp-content'
```

**Version**

- Output of `docker version`:

```
Client:
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.15
 Git commit:        633a0ea838f1
 Built:             Thu Dec 19 00:00:00 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.15
  Git commit:       633a0ea838f1
  Built:            Thu Dec 19 00:00:00 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.1.5_catatonit
  GitCommit:       
```

- Output of `docker info`:

```
Client:
 Debug Mode: false

Server:
 Containers: 106
  Running: 1
  Paused: 0
  Stopped: 105
 Images: 110
 Server Version: 19.03.5
 Storage Driver: btrfs
  Build Version: Btrfs v5.4 
  Library Version: 102
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: oci runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: b34a5c8af56e510852c35414db4c1f4fa6172339
 runc version: 3e425f80a8c931f88e6d94a8c831b9d5aa481657
 init version: 
 Security Options:
  apparmor
  seccomp
   Profile: default
 Kernel Version: 5.6.8-1-default
 Operating System: openSUSE Tumbleweed
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 22.86GiB
 Name: xxx
 ID: xxx
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No swap limit support

```

**Comment by genughaben** [comment]
Unfortunately, I cannot help, but have the same error occurring, when I try to install WordPress using the helm chart version 9.03 or upgrading from 7.3.8. 7.3.8 works. The difference seems to be container was previously run as root, and is now as 1001. Maybe this is a similar underlying issue?

**Comment by genughaben** [comment]
Speaking of that, I was able to fix the error, when I modified the owner, group for the folder that volume uses on the host system.

`sudo chown 1001:1001 /tmp/wp `

I hope that helps you too!

**Comment by Neikius** [comment]
Yeah thanks, I figured it might be something like this ... 
Not sure whether I will use this, wait or just use another image since this image also presents other problems for my case (namely I want to migrate an old install and this doesn't put everything in the volume anyway).

**Comment by marcosbc** [comment]
@genughaben Assuming you're using the chart, by specifying the following it *should* work so it runs as a root user:

    --set securityContext.runAsUser=0 --set securityContext.fsGroup=0

It worked for us:

```console
$ k exec -ti mywordpress-7bb8bddbc4-kvbpm bash
root@mbwordpress-7bb8bddbc4-kvbpm:/# whoami
root
```

The other alternative would be to configure an initContainer that configures volume permissions. This is usually the case when the PVC is based on NFS, maybe it is your case

Note that we already support that for other images, i.e. [MariaDB](https://github.com/bitnami/charts/blob/master/bitnami/mariadb/README.md#parameters) (refer to volumePermissions parameters). If you wish, you could send a PR, we would be happy to review it.

___

@Neikius We have documented the change in the README, and how to workaround it if it does not work for you. If you prefer to use the old approach, just set the "securityContext" parameters during the Helm install/deployment above and it should work.

**Comment by Neikius** [comment]
@marcosbc Sorry but I am confused. I've read the README and the part which mentions I can mount the volume says nothing about this. Yes, a careful reader will find some details in the "changes" section, but that is not something one reads when just trying out something. Also not using k8s/helm so that part does not apply to me. I was actually just copying stuff from examples provided in the README and that failed (see commands I used above). Maybe you could add the appropriate commands to the example so it works? The section is "Persisting your application" then "Mount host directories as data volumes with Docker Compose"

**Comment by marcosbc** [comment]
In this specific case, it was documented in the "Notable changes" section for the container: https://github.com/bitnami/bitnami-docker-wordpress#532-debian-10-r30

> The WordPress container has been migrated to a "non-root" user approach. Previously the container ran as the root user and the Apache daemon was started as the daemon user. From now on, both the container and the Apache daemon run as user 1001. You can revert this behavior by changing `USER 1001` to `USER root` in the Dockerfile.

If you are using the container, and not a chart, you have a few ways to fix this:

* Use a non-root user fixing the permissions before mounting the folder (`sudo chown 1001 /path/to/wordpress-persistence` outside of the container)
* Use a non-root user, fixing permissions with a container (not documented)
* Use a root user, extending the Dockerfile to specify `USER root` or modify `docker-compose.yml`:

```diff
@@ -9,6 +9,7 @@
       - MARIADB_DATABASE=bitnami_wordpress
       - ALLOW_EMPTY_PASSWORD=yes
   wordpress:
+    user: root
     image: 'bitnami/wordpress:5'
     ports:
       - '80:80'
```

But you're right, the examples are currently not working. We will update our documentation to include the following note below the code snippets:

>NOTE: As this is a non-root container, the mounted files and directories must have the proper permissions for the UID `1001`.

**Comment by Sfaraaah** [comment]
I have so many bugs and issues and every time I submit a complaint somebody tells me something else like right now I’m having so many issues and every time I submit a claim someone tells me something else we need to solve my issue Facebook apps and games,  Somehow my personal account ended up being a business account but my accounts were hacked it took me seven days to fix them but I don’t think everything was fixed I think it’s still somehow under a business account and it should be a personal account

**Comment by VladimirAkopyan** [comment]
@genughaben I have encountered this error, but after some mucking about I have discovered that it was caused by local-path-provisioner, not by the wordpress image. Removing the subpath section from the wordpress volume mount fixed the issue for me. If you are using Local-path-provisioner or RKE kubernetes, see below for details.

https://github.com/rancher/local-path-provisioner/issues/4
https://github.com/rancher/rancher/issues/14836
